ryven/main/Ryven.py

```diff
@@ -1,105 +1,559 @@
+
+        sw = StartupDialog(vars(args), parent=gui_parent)
+        # Exit if dialog couldn't initialize or is exited
+        if sw.exec_() <= 0:
+            sys.exit('Start-up screen dismissed')
 
-        # exit if dialog couldn't initialize
-        if sw.editor_startup_configuration == {}:
-            sys.exit()
+        # Update `args` with `sw.configs`
+        for key, value in sw.configs.items():
+            # A little safeguard
+            if hasattr(args, key):
+                setattr(args, key, value)
+            else:
+                raise KeyError(
+                    f'The startup dialog set an unknown argument. '
+                    f'Got: {key}={value}')
 
-        window_theme = sw.window_theme
-        editor_init_config = sw.editor_startup_configuration
     else:
-        window_theme = apply_stylesheet(window_theme_name)
-        editor_init_config = {'action': None}
+        # This is needed, because the stylesheet is applied in `StartupDialog`
+        args.window_theme = apply_stylesheet(args.window_theme)
 
-    # adjust flow theme if not set
-    if flow_theme is None:
-        flow_theme = 'pure dark' if window_theme.name == 'dark' else 'pure light'
+    # Get packages required by the project
+    if args.project:
+        nodes, nodes_not_found, project_dict = process_nodes_packages(
+            args.project, requested_nodes=args.nodes)
+
+        if nodes_not_found:
+            sys.exit(
+                f'The package{"s" if len(nodes_not_found)>1 else ""} '
+                f''''{"', '".join([str(p) for p in nodes_not_found])}' '''
+                f'{"were" if len(nodes_not_found)>1 else "was"} requested, '
+                f'but {"they are" if len(nodes_not_found)>1 else "it is"} not available.'
+                f'\n'
+                f'Update the project file or supply the missing package{"s" if len(nodes_not_found)>1 else ""} '
+                f''''{"', '".join([p.name for p in nodes_not_found])}' '''
+                f'on the command line with the "-n" switch.')
+
+        editor_config['action'] = 'open project'
+        editor_config['requested packages'] = args.nodes
+        editor_config['required packages'] = nodes
+        editor_config['content'] = project_dict
+
+    else:
+        editor_config['action'] = 'create project'
+        editor_config['requested packages'] = args.nodes
+
+    # Adjust flow theme if not set
+    if args.flow_theme is None:
+        if args.window_theme.name == 'dark':
+            args.flow_theme = 'pure dark'
+        else:
+            args.flow_theme = 'pure light'
 
-    # init main console
-    console_stdout_redirect, \
-    console_errout_redirect = init_main_console(window_theme)
+    #
+    # Start and run application
+    #
 
-    # init main window
-    editor = MainWindow(editor_init_config, window_title, window_theme, flow_theme, parent=gui_parent)
+    # FIXME? `window_theme` is of type `WindowTheme`, but `flow_theme` is a `str`
```

Comment:
we *could* rename it to `flow_theme_name` if we want to be really precise, otherwise I'm fine with `flow_theme` for now because ryvencore-qt's API accepts theme names instead of themes (the actual theme classes are only used inside ryvencore-qt)

Reply:
Indeed, but `flow_theme_name` was far too bulky to me, so I just used `flow_theme`.

I was just wondering, if it would be possible to pass the `window_theme` as a string or `flow_theme` as a `FlowTheme` object …

Reply:
I don't see a point in exposing `FlowTheme` classes here, but `WindowTheme` is defined in and for Ryven, so to me it seems intuitive that they are of different types, I don't see a major issue there.

Reply:
Fair enough!